tools/rewriter/DetermineAbi.cpp

```diff
@@ -157,18 +168,22 @@ abiSlotsForArg(const clang::QualType &qt,
         dyn_cast<llvm::StructType>(argInfo.getCoerceToType());
     if (STy) {
       if (argInfo.getCanBeFlattened()) {
+        // A struct is "flattenable" if its individual elements can be passed as arguments.
+        // In this case we classify each element of the struct and add each to the list.
         auto elems = STy->elements();
         std::vector<CAbiArgKind> out = {};
         for (const auto &elem : elems) {
           out.push_back(classifyLlvmType(*elem));
         }
         return out;
       } else {
+        // TODO why a single int slot when not flattenable?
         return {CAbiArgKind::Integral}; // one pointer in register, see `clang's
                                         // ClangToLLVMArgMapping::construct`
       }
     }
     // if not flattenable, classify the single-register value
+    // TODO this seems to be the no-StructType case, not the not-flattenable case
     return classifyDirectType(*qt.getCanonicalType(), astContext);
```

Comment:
I think you're right here; the comment should be corrected to say that we have a scalar type here.